source/game.figure.base.bmx

```diff
@@ -407,68 +405,108 @@ Type TFigureBase extends TSpriteEntity {_exposeToLua="selected"}
 
 
 	Method GetTarget:TFigureTargetBase()
-		if figureTargets.length = 0 then return Null
-		return figureTargets[0]
+		Local target:TFigureTargetBase
+
+		LockMutex(figureTargetsMutex)
```

Comment:
Es sieht so aus, als würde diese Änderung nicht reichen. Im gestrigen Testspiel gab es einen Absturz wegen Nullzugriff in isControllable() - dort wird zweimal hintereinander GetTarget() aufgerufen und beim zweiten Aufruf war es dann null. Vermutlich sollte also auch an solchen Stellen eine lokale Variable zwischengespeichert werden.